src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -0,0 +1,130 @@
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient.Contract;
+
+namespace SonarLint.VisualStudio.TypeScript.Analyzer
+{
+    [Export(typeof(IAnalyzer))]
+    internal class JavaScriptAnalyzer : IAnalyzer
+    {
+        private readonly IEslintBridgeClient eslintBridgeClient;
+        private readonly IEslintBridgeIssueConverter issuesConverter;
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public JavaScriptAnalyzer(IEslintBridgeClient eslintBridgeClient, ILogger logger)
+            : this(eslintBridgeClient, new EslintBridgeIssueConverter(), logger)
+        {
+        }
+
+        internal JavaScriptAnalyzer(IEslintBridgeClient eslintBridgeClient, IEslintBridgeIssueConverter issuesConverter, ILogger logger)
+        {
+            this.eslintBridgeClient = eslintBridgeClient;
+            this.issuesConverter = issuesConverter;
+            this.logger = logger;
+        }
+
+        public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
+        {
+            return languages.Contains(AnalysisLanguage.Javascript);
+        }
+
+        public void ExecuteAnalysis(string path,
+            string charset,
+            IEnumerable<AnalysisLanguage> detectedLanguages,
+            IIssueConsumer consumer,
+            IAnalyzerOptions analyzerOptions,
+            CancellationToken cancellationToken)
+        {
+            Debug.Assert(IsAnalysisSupported(detectedLanguages));
+
+            ExecuteAnalysis(path, consumer, cancellationToken).Forget(); // fire and forget
+        }
+
+        internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, CancellationToken cancellationToken)
+        {
+            // Switch to a background thread
+            await TaskScheduler.Default;
+
+            try
```

Comment:
In [CLangAnalyzer](https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/CFamily/CLangAnalyzer.cs#L132) we have a status bar notifier but I think it should move to AnalyzerController in some form. 
This way all the analyzers could benefit from it, instead of repeating it in the new JS and TS analyzers.

Reply:
I'd create a ticket for the status notification and leave it until later. I don't think we'd delay shipping for that.
However, we need at least a minimal UX to indicate something is happening, so we should log something.

Reply:
Done #2280